### PR TITLE
add option to vagrant provisioning to use stg infra

### PR DESCRIPTION
**Pull request by ryanlerch**
This adds an option when provisioning the bodhi devel
envrionment with vagrant to set up the config to use
the fedora staging infra.

to use this option, the box is provisioned with

    vagrant --use-staging up

note that talking to koji stg requires a krb ticket, so
before starting bodhi, get a new ticket with:

    kinit <fasusername>@STG.FEDORAPROJECT.ORG

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -54,12 +83,19 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  # config.vm.synced_folder ".dnf-cache", "/var/cache/dnf", type: "sshfs", sshfs_opts_append: "-o nonempty"
 
  # Comment this line if you would like to disable the automatic update during provisioning
- config.vm.provision "shell", inline: "sudo dnf upgrade -y"
+ #config.vm.provision "shell", inline: "sudo dnf upgrade -y"
```
Comment (abompard) [inline Vagrantfile]: Is this intended for this change? It seems unrelated.

File: devel/ansible/roles/bodhi/tasks/main.yml [hunk 2]
```diff
@@ -144,6 +144,10 @@
   args:
       creates: /home/vagrant/bodhi/development.ini
 
+- name: Change development.ini to make bodhi use the staging infrastructure
+  import_tasks: configure_stg.yml
+  when: fas is defined
```
Comment (abompard) [inline devel/ansible/roles/bodhi/tasks/main.yml]: Using the `fas` variable for that seems a bit non-intuitive, I think it'd be better if we defined a `staging` variable or a `env` variable and set it to `staging`, and use that here.

**Review by abompard — COMMENTED** [review]
A couple remarks, nothing big.

File: Vagrantfile (hunk 1 above)
Comment (ryanlerch) [reply Vagrantfile]: yup! an unintended change. turned this off during testing. sorry!

File: devel/ansible/roles/bodhi/tasks/main.yml (hunk 2 above)
Comment (ryanlerch) [reply devel/ansible/roles/bodhi/tasks/main.yml]: the fas variable is what is passed into ansible from vagrant (the changes to the vagrantfile now prompts the user for the fas username for the confgiure_stg.yml to add to the config).

I'll pass in the use_staging_infra boolean too, change the fas variable to `staging_fas` and change the when line to 

    when: (staging_fas is defined) and use_staging_infra

for extra clarity here.

does that sound ok?

**Comment by ryanlerch** [comment]
@abompard updated. comments about the fixes are replies to your inline comments.

File: devel/ansible/roles/bodhi/tasks/main.yml (hunk 2 above)
Comment (abompard) [reply devel/ansible/roles/bodhi/tasks/main.yml]: Yes, thanks :-)

File: Vagrantfile (hunk 1 above)
Comment (abompard) [reply Vagrantfile]: No problem, but it seems so be still here...

Comment (ryanlerch) [reply Vagrantfile]: gah, sorry!

i uncommented it, then commented it back out again to speed up testing of the other change, then forgot to uncomment it again! sorry!

just updated the commit again to make sure it is NOT commented out!

**Comment by abompard** [comment]
The CI failure comes from new deprecation warnings in Rawhide that mess up the output, I think they can be ignored for now.

**Review by abompard — APPROVED** [review]
